agent/agentcontainers/api.go

```diff
@@ -403,8 +404,8 @@ func (api *API) Routes() http.Handler {
 	r.Use(ensureInitialUpdateDoneMW)
 
 	r.Get("/", api.handleList)
+	// TODO(mafredri): Simplify this route.
```

Comment:
Maybe note down how exactly you want to simplify for later you (or whoever else it happens to be)